calculator/main.py

```diff
@@ -26,7 +26,20 @@ def perform_operation(calc, operation, params):
     :param params: Tuple containing the list of nums to operate with.
                    ie: (1, 2, 3, 4.5, -2)
     """
-    pass
+    if operation not in calc["operations"]:
+        raise InvalidOperation('"{}" operation not supported.'.format(operation))
+    
+    
+    if params is None or not all([isinstance(param, (float,int)) and type(param) is not bool for param in params]):
+        raise InvalidParams("Given params are invalid.")
+        
```

Comment:
Something like:
type(param) in (int, float) 
should be shorter and more readable.
Try to avoid writing lines that long.
Apart from that your code looks really clean.